src/fmt.ml

```diff
@@ -187,6 +210,23 @@ let styled_string style = styled style string
 let stdout = Format.std_formatter
 let stderr = Format.err_formatter
 
+let with_strfmt f =
+  let buf = Buffer.create 17 in
+  let ppf = Format.formatter_of_buffer buf in
+  f ppf ;
+  Format.pp_print_flush ppf () ;
+  Buffer.contents buf
+
+let strfmt pp_v v = strf "%a" pp_v v
+
+let with_file filename f =
```

Comment:
> Why not, you seem to have fairly similar functions already.

You mean in `Bos` ? Yes, that should be added (but don't care about that at the moment things there need to be redesigned there).